pkg/tfbridge/detailed_diff.go

```diff
@@ -149,6 +152,37 @@ func (differ detailedDiffer) getEffectiveType(path walk.SchemaPath) shim.ValueTy
 	return tfs.Type()
 }
 
+type hashIndexMap map[int]int
+
+func (differ detailedDiffer) calculateSetHashIndexMap(path propertyPath, listVal resource.PropertyValue) hashIndexMap {
```

Comment:
```suggestion
// calculateSetHashIndexMap maps hash values to their respective indices in listVal.
//
// Conceptually, calculateSetHashIndexMap computes:
//
//	m := make(hashIndexMap, len(listVal))
//	for i, v := range listVal {
//		m[hash(v)] = i
//	}
//	return m
//
// listVal must represent an array.
func (differ detailedDiffer) calculateSetHashIndexMap(path propertyPath, listVal resource.PropertyValue) hashIndexMap {
```

Actually, why can't `listVal` be a `[]resource.PropertyValue`? Is that because of `MaxItemsOne: true`?

Reply:
> Actually, why can't listVal be a []resource.PropertyValue? Is that because of MaxItemsOne: true?

We only call it on `resource.NewArrayProperty(...)`. I think we can just change the type to `[]resource.PropertyValue`.